sdcm/kafka/kafka_producer.py

```diff
@@ -0,0 +1,476 @@
+            'bootstrap.servers': ','.join(self.kafka_addresses),
+            'group.id': options.get('group.id', 'validator-group'),
+            'auto.offset.reset': options.get('auto.offset.reset', 'earliest'),
+        }
+        consumer_config.update(options.get('consumer-property', {}))
+
+        self.consumer = Consumer(consumer_config)
+        self.consumer.subscribe([self.topic])
+
+    def run(self) -> None:
+        self.read_kafka_messages()
+        self.read_rows_from_scylla()
+        self.compare_messages_and_rows()
+
+    def read_kafka_messages(self, idle_timeout: int = 10) -> None:
+        LOGGER.info("Consuming messages from Kafka topic %s.", self.topic)
+        last_message_time = time.time()
+        try:
+            while True:
+                msg = self.consumer.poll(1.0)
+                current_time = time.time()
+                if msg is None:
+                    if current_time - last_message_time >= idle_timeout:
+                        LOGGER.debug("No new messages received for %s seconds. Stopping consumption.", idle_timeout)
+                        break
+                    continue
+                if msg.error():
+                    LOGGER.error("Consumer error: %s.", msg.error())
+                    continue
+
+                key = self.key_deserializer(
+                    msg.key(), SerializationContext(self.topic, MessageField.KEY)
+                ) if msg.key() else None
+                value = self.value_deserializer(
+                    msg.value(), SerializationContext(self.topic, MessageField.VALUE)
+                ) if msg.value() else None
+
+                if self.message_format == 'json-with-schema':
+                    key_dict = None
+                    try:
+                        key_dict = json.loads(key)
+                    except json.JSONDecodeError as e:
+                        LOGGER.error("Failed to decode key JSON: %s", e)
+                    value_dict = None
+                    try:
+                        value_dict = json.loads(value)
+                    except json.JSONDecodeError as e:
+                        LOGGER.error("Failed to decode value JSON: %s", e)
+                elif self.message_format == 'avro':
+                    # AVRO messages are already deserialized into dicts
+                    key_dict = key
+                    value_dict = value
+
+                self.collected_messages.append({'key': key_dict, 'value': value_dict})
+                LOGGER.debug("Consumed message: key=%s, value=%s", key_dict, value_dict)
+                last_message_time = current_time
+        finally:
+            self.consumer.close()
+
+    def read_rows_from_scylla(self) -> None:
+        LOGGER.info("Reading rows from ScyllaDB table '%s.%s'.", self.keyspace, self.table)
+        with self.tester.db_cluster.cql_connection_patient(self.tester.db_cluster.nodes[0]) as session:
+            query = f"SELECT * FROM {self.keyspace}.{self.table}"
```

Comment:
for scaling, we'll need to read page by page

Reply:
Added reading of batches of rows in scylla.
Also as was discussed, added seeded generation of data to be written to Kafka; and later when we validate data in scylla we do not read it from kafka, but regenerate using the known seeds.